### PR TITLE
Issue/184/helm integration

### LINKED ISSUE #184: Manifest applier needs to support Helm charts
Flux seems to have good abstractions for this.

Other option is to bundle Helm and use it through golang API.

Probably needs #182 first though

### PR DESCRIPTION
**Pull request by soider**
---
name: Pull Request
about: Create a Pull Request
title: Helm based addons
labels: 
assignees: ''

---

**Issue**
Fixes #184 

**What this PR Includes**
Implementation of the helm based addons for the k0s cluster

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -178,10 +191,31 @@ telemetry:
...
+    url: https://prometheus-community.github.io/helm-charts
+  charts:
+  - name: prometheus-stack
+    chartname: prometheus-community/prometheus
+    version: "11.16.8"
+    values: |2
+        values: "for overriding"
+    namespace: default
+```
+
```
Comment (jnummelin) [inline docs/configuration.md:213]: I think we should also document the CRD structure and how we pick those up from `kube-system` namespace.

File: pkg/apis/v1beta1/cluisterhelmaddons.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
+package v1beta1
```
Comment (jnummelin) [inline pkg/apis/v1beta1/cluisterhelmaddons.go:1]: Needs Apache 2 "header"

**Review by jnummelin — CHANGES_REQUESTED** [review]
Couple minor change requests. Other than that fantastic stuff.

File: cmd/server.go [hunk 3]
```diff
@@ -279,6 +280,13 @@ func createClusterReconcilers(clusterConf *config.ClusterConfig) map[string]comp
 
 	initNetwork(reconcilers, clusterConf)
 
+	manifestsSaver, err := server.NewManifestsSaver("helm")
+	if err != nil {
+		logrus.Warnf("failed to initialize reconcilers manifests saver: %s", err.Error())
+	}
+	reconcilers["crd"] = server.NewCRD(manifestsSaver)
```
[the remark this answers is not in the document]
Comment (soider) [reply cmd/server.go:287]: yes

File: .golangci.yml [hunk 4]
```diff
@@ -2,8 +2,9 @@
 run:
 
   timeout: 5m
-  skip-dirs-use-default: true
-
+  skip-dirs-use-default: false
+  skip-dirs:
+    - pkg/helm
```
Comment (jnummelin) [inline .golangci.yml:7]: Why are we skipping this? Is it because generated code? If so, I'd appreciate a comment here on the reasons.

**Review by jnummelin — APPROVED** [review]

**Pull request merged**